### PR TITLE
fix resource comparasion bug in task topology

### PR DESCRIPTION
**Pull request by Thor-wl**
refer: https://github.com/volcano-sh/volcano/issues/1525
Signed-off-by: Thor-wl <13164644535@163.com>

### CODE REVIEW

File: pkg/scheduler/api/resource_info.go [hunk 1]
```diff
@@ -368,6 +368,35 @@ func (r *Resource) LessEqualStrict(rr *Resource) bool {
 	return true
 }
 
+// LessInSomeDimension returns true if there exists any dimension whose resource amount in r is less than that in rr.
+// Otherwise returns false.
+// Note: Any dimension of resource, which is not listed in resource object, is regarded as zero.
+func (r *Resource) LessInSomeDimension(rr *Resource) bool {
```
Comment (wpeng102) [inline pkg/scheduler/api/resource_info.go:374]: please add some unit-test case for this function

Comment (Thor-wl) [reply pkg/scheduler/api/resource_info.go:374]: OK, done.

**Review by jasonliu747 — COMMENTED** [review]
/lgtm
have you tested locally?

**Review by wpeng102 — APPROVED** [review]
/lgtm

**Review by shinytang6 — APPROVED** [review]
/lgtm
/approve

**Pull request merged**